core/lib/spree/core/product_duplicator.rb

```diff
@@ -70,5 +70,12 @@ def reset_properties
         end
       end
     end
+
+    def generate_new_sku(sku)
+      loop do
+        sku = "COPY OF #{sku}"
```

Comment:
@JDutil I don't understand your doubt, the method `generate_new_sku` uses always the same input parameter `sku` that we don't need it later. This method just returns an sku which does not exist yet in the db, by concatenating "COPY OF" string in each iteration. The timestamp is a good idea too, moreover it must control the  SKUs for master and all variants anyway because they would be created almost in the same time actually.